core/src/ids.rs

```diff
@@ -224,41 +225,63 @@ impl ReservationId {
     }
 }
 
+pub trait ProgramNonce {
```

Comment:
Can't see any reason for the trait and impls for mid, pid and u32/64

Reply:
The idea was to limit the nonce variable to only be one of the three types, but this could be removed and a couple extra functions could be added for the u32 and u64 case. This is only an issue since u32 and u64 don't implement something like AsRef<[u8]>